io-engine/src/grpc/v1/replica.rs

```diff
@@ -468,14 +547,16 @@ impl ReplicaRpc for ReplicaService {
                         };
                     // create snapshot
                     match lvol.create_snapshot(snap_config.clone()).await {
-                        Ok(()) => {
-                            info!("Create Snapshot Success for lvol: {lvol:?}");
+                        Ok(Some(snap_lvol)) => {
+                            info!("Create Snapshot Success for {lvol:?}, {snap_lvol:?}");
                             Ok(CreateReplicaSnapshotResponse {
-                                snapshot_uuid: snap_config
-                                    .txn_id()
-                                    .unwrap_or_default(),
+                                replica_uuid: lvol.uuid(),
+                                snapshot: Some(ReplicaSnapshot::from(snap_lvol)),
                             })
                         }
+                        Ok(None) => {
```

Comment:
Why do we return an Option?

Reply:
It will be handled based on decision about separte function for local and remote snapshot. Currently We wanted to keep it as a single function with Option as parameter.

Reply:
IMHO with a single fn this api can be a bit confusing, as it's not obvious that this is the behaviour.

Reply:
Changed to 2 separate function and Option parameter removed to simplify the return handling.